### ISSUE TITLE
generate past court reports associated with past court dates

### ISSUE THREAD

**Issue by Zrrrpy** [rubyforgood/casa#1741]
Part of epic #1730, dependent on #1739

AS a **supervisor/admin**, 
I WANT to generate past court reports from previous court dates,
SO THAT I can easily reference this information without having to obtain it through another source (such as digging through email, asking the assigned volunteer, etc). 

### How it works now
There is currently no place for generating past court reports.

### How it should work
Once #1739 is complete, **supervisors** and **admins** (and later, volunteers) should be able to generate and download a court report associated with a past court date from the same place where they see past court details.

### Acceptance criteria
- [ ] When supervisors/admins click on a "past court date" on the Edit CASA Case page and see past court details, there should be a button to "generate court report". 
- [ ] Clicking this button should generate a court report associated with the corresponding court date. Once the court report is generated, a button to "download" the court report should appear. _(Sign into QA or staging as a volunteer and click "generate court report" in the lefthand sidebar menu to see how this is currently handled and should be modeled off of.)_
- [ ] Clicking the download button downloads the generated court report.

**Comment by Zrrrpy** [comment]
@khiga8 Since (if my memory serves me correctly!) we were corresponding about downloading past court reports in the past (on another issue), I wanted to tag you here and just let you know this is the direction we're ultimately going in due to some additional changes to court reports and court details. 😄

**Comment by rhian-cs** [comment]
Hey @Zrrrpy, i'll work on this issue. I have some questions before starting, though:
- Should the past court report be associated with a single past court date or with all past court dates of a given CASA case?
- Just to make sure, should the report format be .docx?
- The data present in the past court report should be the court details added in #1739 (date, hearing type, judge and court mandates), is that correct?

**Comment by compwron** [comment]
@Zrrrpy is not currently active on this project so I'll answer :) 
FYI @talya19 these are PM questions

1. past court report should be associated with a single past court date of a case
1. yes, report format should be docx. for *past* reports, maybe we could save them as pdf... but we already do docx so let's just stick with that. 
1. correct, the past report is just a saved copy of the current report with all the case contacts, mandates etc.
